### ISSUE TITLE
Add support for number format expression

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#14842]
This issue tracks adding [number-format](https://docs.mapbox.com/mapbox-gl-js/style-spec/#expressions-types-number-format) expression to the android binding. Core PR is happening in https://github.com/mapbox/mapbox-gl-native/pull/14840, original core issue in https://github.com/mapbox/mapbox-gl-native/issues/13632 (which is a gl-js parity issue to https://github.com/mapbox/mapbox-gl-js/pull/7626).

**Comment by 1ec5** [comment]
Besides adding `number-format` to the Android expression DSL, the actual formatting should be implemented in Java in platform/android/src/string_util.cpp based on [`NumberFormat.format(double)`](https://developer.android.com/reference/java/text/NumberFormat.html#format%28double%29).

/ref https://github.com/mapbox/mapbox-gl-native/pull/14840#pullrequestreview-245848617

**Comment by tobrun** [comment]
Started to work on this, have a basic version working though it doesn't route through the java formatting yet. The following expression will result in a symbol layer as shown in screenshot:

```java
numberFormatSymbolLayer.setProperties(
  textField(
    numberFormat(123.456789, locale("nl"), currency("EUR"))
  )
);
```

![image (9)](https://user-images.githubusercontent.com/2151639/63348256-31694900-c359-11e9-8a44-d2d1a08a5dfc.png)

To be investigated why the € sign isn't showing properly though.

**Comment by tobrun** [comment]
Capturing from @jmalanen that the correct locale for this configuration = `nl-NL`

```java
    numberFormatSymbolLayer.setProperties(
      textField(
        numberFormat(123.456789, locale("nl-NL"), currency("EUR"))
      )
    );

```

![image](https://user-images.githubusercontent.com/2151639/63350056-ff59e600-c35c-11e9-8f87-0d50bd309d62.png)

**Comment by tobrun** [comment]
landed in #15424
